### DISCUSSION TITLE
Dynamically set defaultValues

### DISCUSSION THREAD

**Discussion by JacobWeisenburger** [TanStack/form#613] [Q&A] [answered]
Is there a way to change `defaultValues` in response to clicking a button or something?

Something like this:
```ts
const form = useForm( {
    defaultValues: {
        foo: '',
    },
} )

function onSomeEvent () {
    form.setDefaultValues( { foo: 'new value' } )
}
```

I'm asking this because I would like some way to know if a field is dirty or not and I would like to have a way when I save the form to update the `defaultValues` so those fields are no longer considered dirty

**Comment by fourgates** [comment]
pretty sure this does the trick

```
this.form.update({defaultValues: {foo: 'bat'}})
```

**Reply by crutchcorn** [reply]
Wait, does this work in React after re-rendering the component?

**Comment by crutchcorn** [comment, marked as answer]
Tracking here: https://github.com/TanStack/form/issues/785
